backend/rag_optimizers.py

```
@@ -54,13 +58,16 @@ def rewrite_query(query: str, provider: str, api_key: str, model_path: str = "")
         elapsed = time.time() - start_time
-        print(f"[RAG OPTIMIZER] Optimized to: '{rewritten}' (took {elapsed:.2f}s)")
-        
-        # Cache it
+        logger.debug("Query rewritten to: '%s' (%.2fs)", rewritten, elapsed)
+
+        # Evict oldest entry if cache is full, then store
+        if len(_QUERY_REWRITE_CACHE) >= _CACHE_MAX:
+            oldest_key = next(iter(_QUERY_REWRITE_CACHE))
+            del _QUERY_REWRITE_CACHE[oldest_key]
         _QUERY_REWRITE_CACHE[query] = rewritten
         return rewritten
```

Ruff (0.15.20)

[warning] Consider moving this statement to an `else` block

(TRY300)